tedana/tests/test_integrity _robustica.py

```diff
@@ -0,0 +1,667 @@
+    If current data is not already available, downloads tar.gz data
+    stored at `https://osf.io/osf_id/download`.
+
+    and unpacks into `out_path`.
+
+    Parameters
+    ----------
+    osf_id : str
+       The ID for the OSF file.
+    out_path : str
+        Path to directory where OSF data should be extracted
+    """
+
+    try:
+        datainfo = requests.get(f"https://osf.io/{osf_id}/metadata/?format=datacite-json")
+    except Exception:
+        if len(os.listdir(test_data_path)) == 0:
+            raise ConnectionError(
+                f"Cannot access https://osf.io/{osf_id} and testing data " "are not yet downloaded"
+            )
+        else:
+            TestLGR.warning(
+                f"Cannot access https://osf.io/{osf_id}. "
+                f"Using local copy of testing data in {test_data_path} "
+                "but cannot validate that local copy is up-to-date"
+            )
+            return
+    datainfo.raise_for_status()
+    metadata = json.loads(datainfo.content)
+    # 'dates' is a list with all udpates to the file, the last item in the list
+    # is the most recent and the 'date' field in the list is the date of the last
+    # update.
+    osf_filedate = metadata["dates"][-1]["date"]
+
+    # File the file with the most recent date for comparision with
+    # the lsst updated date for the osf file
+    if os.path.exists(test_data_path):
+        filelist = glob.glob(f"{test_data_path}/*")
+        most_recent_file = max(filelist, key=os.path.getctime)
+        if os.path.exists(most_recent_file):
+            local_filedate = os.path.getmtime(most_recent_file)
+            local_filedate_str = str(datetime.fromtimestamp(local_filedate).date())
+            local_data_exists = True
+        else:
+            local_data_exists = False
+    else:
+        local_data_exists = False
+    if local_data_exists:
+        if local_filedate_str == osf_filedate:
+            TestLGR.info(
+                f"Downloaded and up-to-date data already in {test_data_path}. Not redownloading"
+            )
+            return
+        else:
+            TestLGR.info(
+                f"Downloaded data in {test_data_path} was last modified on "
+                f"{local_filedate_str}. Data on https://osf.io/{osf_id} "
+                f" was last updated on {osf_filedate}. Deleting and redownloading"
+            )
+            shutil.rmtree(test_data_path)
+    req = requests.get(f"https://osf.io/{osf_id}/download")
+    req.raise_for_status()
+    t = tarfile.open(fileobj=GzipFile(fileobj=BytesIO(req.content)))
+    os.makedirs(test_data_path, exist_ok=True)
+    t.extractall(test_data_path)
+
+
+def reclassify_raw() -> str:
+    test_data_path, _ = data_for_testing_info("three-echo-reclassify")
+    return os.path.join(test_data_path, "TED.three-echo")
+
+
+def reclassify_raw_registry() -> str:
+    return os.path.join(reclassify_raw(), "desc-tedana_registry.json")
+
+
+def guarantee_reclassify_data() -> None:
```

Comment:
```suggestion
def guarantee_reclassify_data() -> str:
```

Type hint indicates no return data, but code suggests a string return.